File: Bubble_Sort/Java/pravsingh/bubble_sort.java
package github_contributions;

/**
 * 
 * @author pravsingh
 * 
 */
public class bubble_sort {

	/**
	 * Sorts an integer array in ascending order.
	 * 
	 * @param data
	 *            - reference to the integer array to sort, must not be null
	 *            Postcondition: The array is sorted in ascending order.
	 */
	public static void bubbleSort(int[] data) {

		for (int k = 0; k < data.length - 1; k++) {
			boolean isSorted = true;

			for (int i = 1; i < data.length - k; i++) {
				if (data[i] < data[i - 1]) {
					int tempVariable = data[i];
					data[i] = data[i - 1];
					data[i - 1] = tempVariable;

					isSorted = false;

				}
			}

			if (isSorted)
				break;
		}
	}

}

File: Bubble_Sort/Python/aesptux/bubble_sort.py
def bubble_sort(arr):
    is_sorted = False
    while not is_sorted:
        is_sorted = True
        for i in range(len(arr)-1):
            if arr[i] > arr[i + 1]:
                arr[i], arr[i + 1] = arr[i + 1], arr[i]
                is_sorted = False
